### PR TITLE
Upgrade scalafix, semanticdb, scala

### PR DESCRIPTION
**Pull request by jozic**

### CODE REVIEW

File: src/main/scala/io/github/evis/scalafix/maven/plugin/ScalafixArgumentsBuilder.scala [hunk 1]
```diff
@@ -133,35 +133,33 @@ object ScalafixArgumentsBuilder {
...
 
     def withScalaVersion(scalaVersion: Option[String]): ScalafixArguments =
-      withOpt(_.withScalaVersion, scalaVersion)
+      withOpt(scalaVersion)(_.withScalaVersion)
 
-    private def withOpt[A](
-        f: ScalafixArguments => A => ScalafixArguments,
-        value: Option[A]
-    ): ScalafixArguments =
+    private def withOpt[A](value: Option[A])(
```
Comment (jozic) [inline src/main/scala/io/github/evis/scalafix/maven/plugin/ScalafixArgumentsBuilder.scala:161]: i had to flip arguments order, to help compiler properly resolve overloaded methods (like `withToolClasspath`

**Comment by jozic** [comment]
@evis please check this out

File: pom.xml [hunk 2]
```diff
@@ -11,7 +11,7 @@
     <url>https://github.com/evis/scalafix-maven-plugin</url>
     <groupId>io.github.evis</groupId>
     <artifactId>scalafix-maven-plugin</artifactId>
-    <version>0.1.3_0.9.5</version>
+    <version>0.1.3_0.9.23</version>
```
Comment (evis) [inline pom.xml:14]:
```suggestion
    <version>0.1.4_0.9.23</version>
```

Also, would be great to fix versions in README :)

(can do it myself, though)

**Review by evis — COMMENTED** [review]
Thanks for your contribution! Will merge & release it tomorrow or on Monday

File: pom.xml (hunk 2 above)
Comment (jozic) [reply pom.xml]: changed the version and updated README

**Comment by evis** [comment]
Have you tested these changes on an actual project? I tried to run plugin built from this branch on my local project, and it failed with `A required class was missing while executing io.github.evis:scalafix-maven-plugin:0.1.4_0.9.23:scalafix: scala/reflect/internal/util/ReusableInstance$`.

Full stack trace can be found here: https://gist.github.com/evis/fadab0e2c6ce80803c2998c0d3a11d45

I should add integration tests... :eyes:

**Comment by jozic** [comment]
I actually tried that on my projects and it worked properly, will try again soon

**Comment by evis** [comment]
That was just a bug with installing semanticdb in my project. After fixing it your update works like a charm, thank you! Gonna release it soon.

**Pull request merged**